pyodide_build/pywasmcross.py

```diff
@@ -44,6 +44,13 @@
 symlinks = set(["cc", "c++", "ld", "ar", "gcc", "gfortran"])
 
 
+class EnvironmentRewritingArgument(argparse.Action):
+    def __call__(self, parser, namespace, values, option_string=None):
+        for e_name, e_value in os.environ.items():
+            values = values.replace(f"$({e_name})", e_value)
+        setattr(namespace, self.dest, values)
```

Comment:
Could you please add some tests (outside of pyodide) for this class? in pyodide_built/tests/test_pywasmcross.py

Reply:
Found out about monkeypatch and done this now. Learn a new thing every day!